source/extensions/filters/network/common/redis/client_impl.cc

```diff
@@ -167,7 +169,9 @@ void ClientImpl::onConnectOrOpTimeout() {
     host_->stats().cx_connect_fail_.inc();
   }
 
-  connection_->close(Network::ConnectionCloseType::NoFlush);
+  if (!is_blocking_client_) {
```

Comment:
we can have a debug level log here, just to indicate that the blocking client is ignoring optimeout

Reply:
Sure, will add